components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionFactory.java

```diff
@@ -131,184 +93,160 @@ private void handleException(String msg) {
             log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
         } catch (IOException e) {
-            log.error("[" + name + "] Error creating connection to RabbitMQ Broker. Reattempting to connect.", e);
-            int retryC = 0;
-            while ((connection == null) && ((retryCount == -1) || (retryC < retryCount))) {
-                retryC++;
-                log.info("[" + name + "] Attempting to create connection to RabbitMQ Broker" +
-                        " in " + retryInterval + " ms");
-                try {
-                    Thread.sleep(retryInterval);
-                    connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
-                    log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
-                } catch (InterruptedException e1) {
-                    log.error("[" + name + "] Error while trying to reconnect to RabbitMQ Broker", e1);
-                } catch (IOException e2) {
-                    log.error("[" + name + "] Error while trying to reconnect to RabbitMQ Broker", e2);
-                }
-            }
+            log.error("[" + name + "] Error creating connection to RabbitMQ Broker. " +
+                    "Reattempting to connect.", e);
+            connection = retry(connection);
             if (connection == null) {
-                handleException("[" + name + "] Could not connect to RabbitMQ Broker. Error while creating connection", e);
+                throw new RabbitMQException("[" + name + "] Could not connect to RabbitMQ Broker. " +
+                        "Error while creating connection", e);
             }
         }
         return connection;
     }
 
-    public int getRetryInterval() {
-        return retryInterval;
-    }
-
-    public int getRetryCount() {
-        return retryCount;
+    /**
+     * Retry when could not connect to the broker
+     *
+     * @param connection the failure connection {@link Connection} object
+     * @return the {@link Connection} object after retry completion
+     */
+    private Connection retry(Connection connection) {
+        int retryC = 0;
+        while ((connection == null) && ((retryCount == -1) || (retryC < retryCount))) {
+            retryC++;
+            log.info("[" + name + "] Attempting to create connection to RabbitMQ Broker" +
+                    " in " + retryInterval + " ms");
+            try {
+                Thread.sleep(retryInterval);
+                connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
+                log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
+            } catch (InterruptedException e1) {
+                Thread.currentThread().interrupt();
+            } catch (IOException e1) {
+                log.error("[" + name + "] Error while trying to reconnect to RabbitMQ Broker", e1);
+            }
+        }
```

Comment:
If the exitted condition is because retryC >= retryCount We might need a log mentioning "unable to create a conntection for this many attempts. Hence existting" or something like that?

Reply:
There is a code below in `createConnection` method.

```
    public Connection createConnection() throws RabbitMQException {
        Connection connection = null;
        try {
            connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
            log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
        } catch (IOException e) {
            log.error("[" + name + "] Error creating connection to RabbitMQ Broker. " +
                    "Reattempting to connect.", e);
            connection = retry(connection);
            if (connection == null) {
                throw new RabbitMQException("[" + name + "] Could not connect to RabbitMQ Broker. " +
                        "Error while creating connection", e);
            }
        }
        return connection;
    }
```